pkg/operator/client/client.go

```diff
@@ -296,9 +300,9 @@ func (c *Client) deployHelmCharts(deployArgs operatortypes.DeployAppArgs) (*comm
 	return installResult, nil
 }
 
-func (c *Client) setResults(deployArgs operatortypes.DeployAppArgs, dryRunResult *commandResult, applyResult *commandResult, helmResult *commandResult) error {
+func (c *Client) setResults(deployArgs operatortypes.DeployAppArgs, dryRunResult *commandResult, applyResult *commandResult, helmResult *commandResult) (*DeployResults, error) {
```

Comment:
How about changing return type to `DeployResults`, so it's never nil

Reply:
maybe, i feel `nil` is better than having empty results because it will have `IsError` set to false by default.